controllers/ssp_controller.go

```diff
@@ -680,19 +676,43 @@ func watchClusterResources(builder *ctrl.Builder, sspOperands []operands.Operand
 	)
 }
 
-func watchResources(builder *ctrl.Builder, handler handler.EventHandler, sspOperands []operands.Operand, watchTypesFunc func(operands.Operand) []client.Object, hookFunc handler_hook.HookFunc) {
-	watchedTypes := make(map[reflect.Type]struct{})
+func watchResources(ctrlBuilder *ctrl.Builder, handler handler.EventHandler, sspOperands []operands.Operand, watchTypesFunc func(operands.Operand) []operands.WatchType, hookFunc handler_hook.HookFunc) {
+	// Deduplicate watches
+	watchedTypes := make(map[reflect.Type]operands.WatchType)
 	for _, operand := range sspOperands {
-		for _, t := range watchTypesFunc(operand) {
-			if _, ok := watchedTypes[reflect.TypeOf(t)]; ok {
-				continue
+		for _, watchType := range watchTypesFunc(operand) {
+			key := reflect.TypeOf(watchType.Object)
+			if wt, ok := watchedTypes[key]; ok {
+				watchType.WatchFullObject = watchType.WatchFullObject || wt.WatchFullObject
```

Comment:
Can you place a comment here, what this is for? I assume it should watch the full object as soon as more than one operand defines a Watch for it?

Reply:
Not "more then one operand". If at least one operand wants to watch the full object, then the watch should allow it.

Reply:
Ok, a comment would still be helpful.

Reply:
Done